scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -562,6 +562,18 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
     }
   }
 
+  // corresponding quasiquote (rank 1) for expressions like `{..$smth}`
+  def constructR1Quasi[T <: Tree : AstMetadata](ctor: => Tree): List[T] = {
+    val tree = {  
+      val q = ctor match {
+        case q: Quasi if q.rank == 0 => q
+        case _ => syntaxError("impossible to construct rank-1 quasiquote", at = token)
+      }
+      atPos(q, q)(implicitly[AstMetadata[T]].quasi(q.rank, q.tree))
+    }
+    implicitly[AstMetadata[T]].quasi(1, tree) :: Nil
+  }
+
```

Comment:
So the output of the `constructR1Quasi...` call here is going to be `Case.Quasi(1, Term.Quasi(0, ...))`? Isn't this misleading? Shouldn't the second argument of an 1+ rank quasi be a tree with the same type as the enclosing quasi?